tool/gravity/cli/ops.go

```diff
@@ -81,33 +82,87 @@ func uploadUpdate(ctx context.Context, tarballEnv *localenv.TarballEnvironment,
 		return trace.Wrap(err)
 	}
 
-	appPackage, err := install.GetAppPackage(tarballEnv.Apps)
+	application, err := install.GetApp(tarballEnv.Apps)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	env.PrintStep("Importing application %v v%v", appPackage.Name, appPackage.Version)
-	_, err = appservice.PullApp(appservice.AppPullRequest{
-		SrcPack: tarballEnv.Packages,
-		SrcApp:  tarballEnv.Apps,
-		DstPack: clusterPackages,
-		DstApp:  clusterApps,
-		Package: *appPackage,
+	var registries []string
+	err = utils.Retry(defaults.RetryInterval, defaults.RetryLessAttempts, func() error {
```

Comment:
Frankly i never understood the retryattemps/lessattempts semantics - can we just define the timeout values for that?
 Also, do you think it might make sense to implement it with a backoff?